plugins/wacom-usb/fu-wac-module.c

```diff
@@ -90,6 +90,10 @@ fu_wac_module_refresh_cb(FuDevice *device, gpointer user_data, GError **error)
 		return FALSE;
 	}
 	if (priv->status != FU_WAC_MODULE_STATUS_OK) {
+		/* retry not necessary for unrecoverable errors */
+		if (priv->status != FU_WAC_MODULE_STATUS_BUSY)
+			return TRUE;
```

Comment:
this logic seems wonky. Can't both be top level conditionals? e.g. `if (priv->status != FU_WAC_MODULE_STATUS_OK && priv->status != FU_WAC_MODULE_STATUS_BUSY)`

Reply:
Using the suggested conditional would not work as intended:

| Status | Should Return | Combined Condition Returns | Note |
|-|-|-|-|
|OK|TRUE|TRUE|No error. We should return TRUE stops further retry attempts.|
|BUSY|FALSE|TRUE|Hardware is not ready yet. We should return FALSE to trigger another retry attempt.|
|(other)|TRUE|FALSE|Unrecoverable error. We should return TRUE to stop further retry attempts.|

Moving the unrecoverable status check above the existing check could work; e.g. something like:
~~~patch
@@ -89,6 +89,11 @@ fu_wac_module_refresh_cb(FuDevice *device, gpointer user_data, GError **error)
                g_propagate_error(error, g_steal_pointer(&error_local));
                return FALSE;
        }
+
+       /* retry not necessary for unrecoverable errors */
+       if (priv->status != FU_WAC_MODULE_STATUS_BUSY)
+               return TRUE;
+
        if (priv->status != FU_WAC_MODULE_STATUS_OK) {
                g_set_error(error,
                            FWUPD_ERROR,
~~~

Or a bit more verbose but also possibly a bit easier to read:
~~~patch
@@ -76,6 +76,12 @@ fu_wac_module_refresh(FuWacModule *self, GError **error)
        return TRUE;
 }
 
+static gboolean
+fu_wac_module_status_is_recoverable(guint8 status)
+{
+       return status == FU_WAC_MODULE_STATUS_BUSY;
+}
+
 static gboolean
 fu_wac_module_refresh_cb(FuDevice *device, gpointer user_data, GError **error)
 {
@@ -89,18 +95,23 @@ fu_wac_module_refresh_cb(FuDevice *device, gpointer user_data, GError **error)
                g_propagate_error(error, g_steal_pointer(&error_local));
                return FALSE;
        }
-       if (priv->status != FU_WAC_MODULE_STATUS_OK) {
-               g_set_error(error,
-                           FWUPD_ERROR,
-                           FWUPD_ERROR_INTERNAL,
-                           "refresh returned status 0x%x [%s]",
-                           priv->status,
-                           fu_wac_module_status_to_string(priv->status));
-               return FALSE;
-       }
 
-       /* success */
-       return TRUE;
+       /* success; no retry needed */
+       if (priv->status == FU_WAC_MODULE_STATUS_OK)
+               return TRUE;
+
+       /* unrecoverable error; no retry needed */
+       if (!fu_wac_module_status_is_recoverable(priv->status))
+               return TRUE;
+
+       /* retry recoverable errors */
+       g_set_error(error,
+                   FWUPD_ERROR,
+                   FWUPD_ERROR_INTERNAL,
+                   "refresh returned status 0x%x [%s]",
+                   priv->status,
+                   fu_wac_module_status_to_string(priv->status));
+       return FALSE;
 }
 
 gboolean
~~~

Reply:
the 1st one looks good.

Reply:
Pushing update with 1st version